### PR TITLE
JBEHAVE-1421 Introduce afterScenario(Timing) hook

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/reporters/NullStoryReporter.java [hunk 1]
```diff
@@ -123,6 +124,10 @@ public void scenarioNotAllowed(Scenario scenario, String filter) {
     public void afterScenario() {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/NullStoryReporter.java:128]: deprecate ?

File: jbehave-core/src/test/java/org/jbehave/core/reporters/StoryNarrator.java [hunk 2]
```diff
@@ -183,8 +188,15 @@ static void narrateAnInterestingStoryNotAllowedByFilter(StoryReporter reporter,
...
-            reporter.afterScenario();
+            reporter.afterScenario(getTiming());
         }
         reporter.afterStory(false);
     }
+
+    private static Timing getTiming() {
+        Timing timing = mock(Timing.class);
+        when(timing.getStart()).thenReturn(1l);
+        when(timing.getEnd()).thenReturn(1l);
```
Comment (valfirst) [inline jbehave-core/src/test/java/org/jbehave/core/reporters/StoryNarrator.java]: it's better to use different values for `start` and `end`

File: jbehave-core/src/test/resources/stream-story-custom-patterns.xml [hunk 3]
```diff
@@ -61,6 +61,8 @@
 <outcome><value>A wrong date</value><value>01/01/2011</value><value>&quot;02/01/2011&quot;</value><value>No</value></outcome>
 </outcomes>
 <step outcome="successful">Given a scenario step</step>
+<parameter name="start">1</parameter>
```
Comment (valfirst) [inline jbehave-core/src/test/resources/stream-story-custom-patterns.xml]: `parameter` is ambiguous name, it's better to introduce new tags

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Comment by uarlouski** [comment]
In master https://github.com/jbehave/jbehave-core/commit/351ef231957faed925a6a3d5d5e0449d765cf0d5

**Pull request closed without merging**